app/controllers/catalog_controller.rb

```diff
@@ -112,13 +112,15 @@ def atomic_st_edit
       if !@record.id.nil? && need_prov_dialogs?(@record.prov_type)
         prov_set_form_vars(MiqRequest.find(@record.service_resources[0].resource_id))      # Set vars from existing request
         @edit[:new][:st_prov_type] = @record.prov_type
+        @edit[:new][:generic_subtype] = @record.generic_subtype if @record.try(:id) && @record.prov_type == 'generic'
```

Comment:
@ZitaNemeckova same code is repeated in else block, why not move it outside if/else block